File: src/cshape_code/DesignPattern/SingletonPattern/SingletonObjectB.cs
using System;
using System.Threading;
namespace DesignPattern.SingletonPattern
{
    /// <summary>
    /// Lazy: true
    /// Thread safe: false
    /// </summary>
    public class SingletonObjectB
    {
        public static int count = 0;

        private static SingletonObjectB singletonObjectB;

        private SingletonObjectB()
        {
            Interlocked.Increment(ref count);
        }

        public static SingletonObjectB GetSingletonObjectB()
        {
            if (singletonObjectB == null)
            {
                singletonObjectB = new SingletonObjectB();
            }

            return singletonObjectB;
        }

        public void ShowMessage()
        {
            Console.WriteLine("Singleton Object B");
        }
    }
}

File: src/java_code/singletonpattern/SingletonObjectB.java
package singletonpattern;

public class SingletonObjectB{
    private static SingletonObjectB singletonObjectBInstance;

    private SingletonObjectB(){
    }

    public static SingletonObjectB getSingletonObjectB(){
        if(singletonObjectBInstance == null){
            singletonObjectBInstance = new SingletonObjectB();
        }

        return singletonObjectBInstance;
    }    
}